src/api/jacobian.jl

```diff
@@ -43,21 +43,18 @@ function jacobian{A}(f, ::Type{A}=Void;
                      chunk_size::Int=default_chunk_size,
                      cache::ForwardDiffCache=ForwardDiffCache(),
                      output_length::Int=0)
-    # if output_length > 0, assume that f is of 
-    # the form f!(output, x), and generate the 
+    # if output_length > 0, assume that f is of
+    # the form f!(output, x), and generate the
     # appropriate closure
     if output_length > 0
-        newf = @eval begin
-            output_cache = ForwardDiffCache()
-            @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
-                output = get_workvec!(output_cache, 
-                                      GradientNumber, T,
-                                      Val{$output_length}, 
-                                      Val{N})
-                $f(output, x)
-                return output
-            end
-            return newf
+        output_cache = ForwardDiffCache()
+        @inline function newf{N,T,C}(x::Vector{GradientNumber{N,T,C}})
```

Comment:
Why not? Just replace function call with the body of the inlined function at the call site. Where that function originated from shouldn't change anything, right (my understanding of the whole process is pretty naive)?

Is this one of those weird situations that's hard to compile for (e.g. "the compiler doesn't want to inline functions passed as arguments")?